File: monitoring-server/GeneratorService/Models/ParamValue.cs
using System;

namespace GeneratorService.Models
{
    /// <summary>
    /// Represents value of particular equipment instance param.
    /// </summary>
    public class ParamValue
    {
        /// <summary>
        /// Internal param value identifier.
        /// </summary>
        public long Id { get; set; }
        /// <summary>
        /// External param value identifier.
        /// </summary>
        public Guid ParamValueUid { get; set; }
        /// <summary>
        /// Equipment instance external identifier that value belongs to.
        /// </summary>
        public Guid EquipmentUid { get; set; }
        /// <summary>
        /// Equipment param external identifier that value belongs to.
        /// </summary>
        public Guid ParamUid { get; set; }
        /// <summary>
        /// Param value.
        /// </summary>
        public string Value { get; set; }
    }
}

File: monitoring-server/generator-service/src/main/scala/com/qwerfah/generator/models/ParamValue.scala
package com.qwerfah.generator.models

import java.time.LocalDateTime

import com.qwerfah.common.Uid

/** Equipment model parameter value associated with specific equipment instance.
  * @param id
  *   Parameter value internal identifier.
  * @param uid
  *   Parameter value external identifier.
  * @param paramUid
  *   Uid of associated parameter.
  * @param instanceUid
  *   Uid of associated equipment instance.
  * @param value
  *   Parameter value string representation.
  * @param time
  *   Param adding time.
  */
final case class ParamValue(
  id: Option[Int],
  uid: Uid,
  paramUid: Uid,
  instanceUid: Uid,
  value: String,
  time: LocalDateTime
)
